server/bin/pbench-unpack-tarballs.py

```diff
@@ -77,7 +44,8 @@ def main(cfg_name):
     init_db(config, logger)
 
     # Initiate the unpacking
-    ntotal, nsuccess = unpack_tarball(config, logger)
+    unpack_obj = Unpack(config, _linksrc, logger)
+    ntotal, nsuccess = unpack_obj.unpack_tarball()
 
     result_string = (
         f"Total processed: {ntotal},"
```

Comment:
> Actually using click would probably be overkill as there's no argument parsing.

There is one (hidden) argument -- the Pbench Config file -- and I believe that recasting this as a Click shell would take care of that, so I think it would actually be a worthwhile thing to do.  (As the code currently stands, I think the help text which says `use --config <file> on the command line` is broken -- I don't see any code which processes command line options like `--config`....)

And, I second the idea of moving the report generation out of here and into the `Unpack` object, although I would recommend passing the parameters explicitly rather than the tight coupling which results from depending on them being passed between the methods as object attributes.

Reply:
[Responding to [Dave's reply to the above](https://github.com/distributed-system-analysis/pbench/pull/2975#issuecomment-1234691211) which escaped from this "conversation".]

>>There is one (hidden) argument -- the Pbench Config file

>There really isn't; this is a "cron pipeline" component, which are all constructed with an implicit hard dependency on the crontab file's environment configuration, which provides
>
> `_PBENCH_SERVER_CONFIG=/opt/pbench-server/lib/config/pbench-server.cfg`

Just because the environment variable is provided by the crontab file instead of some other file or mechanism doesn't change the fact that it's a (hidden) parameter to the script!  😀  And, if the script were Click-based, then the crontab could use the `-C` switch on the command invocation...not that we would _want_ to do that, but then it wouldn't be hidden....  😆